rllib/connectors/common/add_observations_from_episodes_to_batch.py

```diff
@@ -136,15 +136,22 @@ def __call__(
         # If "obs" already in data, early out.
         if Columns.OBS in batch:
             return batch
-
-        for sa_episode in self.single_agent_episode_iterator(
-            episodes,
-            # If Learner connector, get all episodes (for train batch).
-            # If EnvToModule, get only those ongoing episodes that just had their
-            # agent step (b/c those are the ones we need to compute actions for next).
-            agents_that_stepped_only=not self._as_learner_connector,
+        for i, sa_episode in enumerate(
+            self.single_agent_episode_iterator(
+                episodes,
+                # If Learner connector, get all episodes (for train batch).
+                # If EnvToModule, get only those ongoing episodes that just had their
+                # agent step (b/c those are the ones we need to compute actions for
+                # next).
+                agents_that_stepped_only=not self._as_learner_connector,
+            )
         ):
             if self._as_learner_connector:
+                # TODO (sven): Resolve this hack by adding a new connector piece that
```

Comment:
I think this is only necessary, if an artificial timestep had to be added to each episode beforehand (b/c of e.g. GAE or vtrace). I don't think we need to do this here in DQN, correct?
What happens if you remove this logic again?

Reply:
Good question. Another test :D